pandas/io/parsers.py

```diff
@@ -801,6 +803,15 @@ def _clean_options(self, options, engine):
                                   " different from '\s+' are"\
                                   " interpreted as regex)"
                 engine = 'python'
+            elif quoting != csv.QUOTE_NONE:
```

Comment:
Unfortunately, it is very different.  Python's native `csv` library does not accept regex, so we have to do that buggy regex split.